packages/flutter/lib/src/widgets/image.dart

```diff
@@ -459,6 +551,182 @@ class Image extends StatefulWidget {
+  ///           child: result,
+  ///         );
+  ///         if (!_controller.isAnimating) {
+  ///           _controller.animateTo(1.0, curve: Curves.easeOut);
+  ///         }
+  ///       }
+  ///       return result;
+  ///     },
+  ///   );
+  /// }
+  /// ```
+  /// {@end-tool}
+  ///
+  /// Run against a real-world image, the previous example renders the following
+  /// image.
+  ///
+  /// {@animation 400 400 https://flutter.github.io/assets-for-api-docs/assets/widgets/frame_builder_image.mp4}
+  final ImageFrameBuilder frameBuilder;
+
+  /// A builder that specifies the widget to display to the user while an image
+  /// is still loading.
+  ///
+  /// If this is null, and the image is loaded incrementally (e.g. over a
+  /// network), the user will receive no indication of the progress as the
+  /// bytes of the image are loaded.
+  ///
+  /// Once an image has fully loaded, this builder will no longer be consulted.
+  ///
+  /// For more information on how to interpret the arguments that are passed to
+  /// this builder, see the documentation on [ImageLoadingBuilder].
+  ///
+  /// ## Performance implications
+  ///
+  /// If a [loadingBuilder] is specified for an image, the [Image] widget is
+  /// likely to be rebuilt on every
+  /// [rendering pipeline frame](rendering/RendererBinding/drawFrame.html) until
+  /// the image has loaded. This is useful for cases such as displaying a loading
+  /// progress indicator, but for simpler cases such as displaying a placeholder
+  /// widget that doesn't depend on the loading progress (e.g. static "loading"
+  /// text), [frameBuilder] will likely work and not incur as much cost.
+  ///
+  /// ## Chaining with [frameBuilder]
+  ///
+  /// If a [frameBuilder] has _also_ been specified for an image, the two
+  /// builders will be chained together, and the `child` argument to this
+  /// builder will contain the _result_ of the [frameBuilder]. For example,
+  /// consider the following builders used in conjunction:
+  ///
+  /// ```dart
+  /// Image(
+  ///   ...
+  ///   frameBuilder: (BuildContext context, Widget child, int frame, bool wasSynchronouslyLoaded) {
+  ///     return Padding(
+  ///       padding: EdgeInsets.all(8.0),
+  ///       child: child,
+  ///     );
+  ///   },
+  ///   loadingBuilder: (BuildContext context, Widget child, ImageChunkEvent loadingProgress) {
+  ///     return Center(child: child);
+  ///   },
+  /// )
+  /// ```
+  ///
+  /// In this example, the widget hierarchy will contain the following:
+  ///
+  /// ```dart
+  /// Center(
+  ///   Padding(
+  ///     padding: EdgeInsets.all(8.0),
+  ///     child: <image>,
+  ///   ),
+  /// )
+  /// ```
+  ///
+  /// {@tool snippet --template=stateless_widget_material}
+  ///
+  /// The following sample uses [loadingBuilder] to show a
+  /// [CircularProgressIndicator] while an image loads over the network.
+  ///
+  /// ```dart
+  /// Widget build(BuildContext context) {
+  ///   return DecoratedBox(
```

Comment:
The example above didn't include this in the code, but I believe the animation showed it. Maybe unify?

Reply:
Done